applications/FluidDynamicsApplication/custom_elements/embedded_fluid_element.cpp

```diff
@@ -0,0 +1,494 @@
+#include "custom_elements/embedded_fluid_element.h"
+#include "custom_elements/qs_vms.h"
+#include "custom_elements/symbolic_navier_stokes.h"
+
+#include "custom_utilities/embedded_data.h"
+#include "custom_utilities/time_integrated_qsvms_data.h"
+#include "custom_utilities/symbolic_navier_stokes_data.h"
+
+#include "modified_shape_functions/triangle_2d_3_modified_shape_functions.h"
+#include "modified_shape_functions/tetrahedra_3d_4_modified_shape_functions.h"
+
+namespace Kratos {
+
+//template class EmbeddedFluidElement< QSVMS< TimeIntegratedQSVMSData<3,4> > >;
+template class EmbeddedFluidElement< SymbolicNavierStokes< SymbolicNavierStokesData<2,3> > >;
+template class EmbeddedFluidElement< SymbolicNavierStokes< SymbolicNavierStokesData<3,4> > >;
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Life cycle
+
+template< class TBaseElement >
+EmbeddedFluidElement<TBaseElement>::EmbeddedFluidElement(IndexType NewId):
+    TBaseElement(NewId)
+{}
+
+template< class TBaseElement >
+EmbeddedFluidElement<TBaseElement>::EmbeddedFluidElement(IndexType NewId, const NodesArrayType& ThisNodes):
+    TBaseElement(NewId,ThisNodes)
+{}
+
+
+template< class TBaseElement >
+EmbeddedFluidElement<TBaseElement>::EmbeddedFluidElement(IndexType NewId, Geometry<NodeType>::Pointer pGeometry):
+    TBaseElement(NewId,pGeometry)
+{}
+
+template< class TBaseElement >
+EmbeddedFluidElement<TBaseElement>::EmbeddedFluidElement(IndexType NewId, Geometry<NodeType>::Pointer pGeometry, Properties::Pointer pProperties):
+    TBaseElement(NewId,pGeometry,pProperties)
+{}
+
+
+template< class TBaseElement >
+EmbeddedFluidElement<TBaseElement>::~EmbeddedFluidElement()
+{}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Public Operations
+
+template< class TBaseElement >
+Element::Pointer EmbeddedFluidElement<TBaseElement>::Create(IndexType NewId,NodesArrayType const& ThisNodes,Properties::Pointer pProperties) const
+{
+    return Element::Pointer(new EmbeddedFluidElement(NewId, this->GetGeometry().Create(ThisNodes), pProperties));
```

Comment:
I tend to use the old `::Pointer` syntax as future-proofing (I guess that removing dependencies to boost implies using C++11 counted pointers here?)

Reply:
this is the reason for which i proposed a 

Kratos::make_shared in #1297

this will be boost::make_shared as long as we use boost,

then we will change it to std::make_shared.

I will not insist further on this, however just beware that make_shared is expected to be faster than the use of new (according to the docs)

Reply:
In this case I prefer to adapt to #1297 once it is merged.